clrenv/lazy_env.py

```diff
@@ -42,26 +44,27 @@ def __getitem__(self, key):
 _env = {}
 def get_env(*mode):
     global _env
+
     if not mode in _env:
         y = (_load_current_environment(),)
         upaths = find_user_environment_paths()
         y = tuple(yaml.load(open(p).read()) for p in upaths if os.path.isfile(p)) + y
 
-        assignments = filter(lambda m: m.find('=') != -1, mode)
-        mode = filter(lambda m: m.find('=') == -1, mode)
+        assignments = tuple(m for m in mode if m.find('=') != -1)
+        mode = tuple(m for m in mode if m.find('=') == -1)
 
         overrides = []
         for a in assignments:
             overrides.append(a.split('=', 1))
 
-        allenvs = set(chain(*(x.keys() for x in y)))
+        allenvs = set(chain(*(list(x.keys()) for x in y)))
```

Comment:
curious why the new `list(...)`s here and below are necessary. does the code break without them?

Reply:
They probably aren't necessary here, but `x.keys()` in Python 3 doesn't return a `list`. In general, this type of conversion prevents bugs from code that depend on `keys` returning a list.

Reply:
understood, but these usages didn't look like they required lists. definitely just curious, not a big deal. no need to change now!

Reply:
Yea, these changes are autogenerated by a script that is trying to preserve behavior in all cases. Once everything works under Python 3 we can revisit and remove unnecessary type conversions.

Reply:
ahh makes sense now! if this is from e.g. `2to3`, i don't care about little things like `list(...)` then. i for one welcome our new robot overlords, and all that.